src/React/SyntheticEvent.purs

```diff
@@ -81,34 +148,27 @@ foreign import data NativeAbstractView :: Type
 
 foreign import data NativeTouchList :: Type
 
-foreign import preventDefault :: forall eff r. Record (SyntheticEvent' r) -> Eff eff Unit
-
-foreign import isDefaultPrevented :: forall eff r. Record (SyntheticEvent' r) -> Eff eff Boolean
-
-foreign import stopPropagation :: forall eff r. Record (SyntheticEvent' r) -> Eff eff Unit
-
-foreign import isPropagationStopped :: forall eff r. Record (SyntheticEvent' r) -> Eff eff Boolean
-
-foreign import persist :: forall eff r. Record (SyntheticEvent' r) -> Eff eff Unit
-
-type SyntheticEvent' r
+type SyntheticEvent' eff r
   = ( bubbles :: Boolean
     , cancelable :: Boolean
     , currentTarget :: NativeEventTarget
     , defaultPrevented :: Boolean
     , eventPhase :: Number
     , isTrusted :: Boolean
     , nativeEvent :: NativeEvent
+    , preventDefault :: Eff eff Unit
+    , isDefaultPrevented :: Eff eff Boolean
+    , stopPropagation :: Eff eff Unit
+    , isPropagationStopped :: Eff eff Boolean
```

Comment:
Is it necessary to add these as rows in the event? All SyntheticEvents support these methods, and if they weren't in the row then we wouldn't need to propagate the `eff` variable.